modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/provider/test/DefaultSegmentsEntryProviderTest.java

```diff
@@ -319,6 +318,37 @@ public void testGetSegmentsEntryIdsWithMultipleModelCriterion()
 				segmentsEntryIds));
 	}
 
+	@Test
+	public void testGetSegmentsEntryIdsWithNonmatchingContextCriterionAndMatchingModelCriterion()
+		throws Exception {
+
+		_user1 = UserTestUtil.addUser(_group.getGroupId());
```

Comment:
maybe we are not deleting the user1?

Reply:
we are, because of 

```
@DeleteAfterTestRun
private User _user1;
```

Reply:
If this variable is global, why do we create it in the test?

Reply:
Because depending on the test scenario It's created in different ways. For example:

In test testGetSegmentsEntryClassPKsWithMultipleCriterion:

```
_user1 = UserTestUtil.addOrganizationUser(organization, RoleConstants.ORGANIZATION_USER);
```

And here testGetSegmentsEntryClassPKsWithMultipleCriterionNotMatching:

`_user1 = UserTestUtil.addUser();
`

Reply:
I'd use a list but with the disadvantage you need to manually insert the entity

Reply:
Ok, looks good to me, we usually use the list, but is ok for this use case.